contracts/compounder/Strategy.sol

```diff
@@ -0,0 +1,175 @@
+
+/**
+ * @title   AuraBalStrategy
+ * @author  llama.airforce -> AuraFinance
+ * @notice  Changes:
+ *          - remove option to lock auraBAL instead of swapping it
+ *          - remove paltform fee
+ */
+contract AuraBalStrategy is Ownable, AuraBalStrategyBase {
+    using SafeERC20 for IERC20;
+
+    address public immutable vault;
+    address[] public rewardTokens;
+    mapping(address => address) public rewardHandlers;
+
+    uint256 public constant FEE_DENOMINATOR = 10000;
+
+    constructor(
+        address _vault,
+        // AuraBalStrategyBase
+        address _balVault,
+        address _auraBalStaking,
+        address _balToken,
+        address _wethToken,
+        address _auraToken,
+        address _auraBalToken,
+        address _bbusdToken,
+        bytes32 _auraBalBalETHBptPoolId,
+        bytes32 _balETHPoolId
+    )
+        AuraBalStrategyBase(
+            _balVault,
+            _auraBalStaking,
+            _balToken,
+            _wethToken,
+            _auraToken,
+            _auraBalToken,
+            _bbusdToken,
+            _auraBalBalETHBptPoolId,
+            _balETHPoolId
+        )
+    {
+        vault = _vault;
+    }
+
+    /// @notice Set approvals for the contracts used when swapping & staking
+    function setApprovals() external {
+        IERC20(AURABAL_TOKEN).safeApprove(address(auraBalStaking), 0);
+        IERC20(AURABAL_TOKEN).safeApprove(address(auraBalStaking), type(uint256).max);
+        IERC20(BAL_TOKEN).safeApprove(address(balVault), 0);
+        IERC20(BAL_TOKEN).safeApprove(address(balVault), type(uint256).max);
+        IERC20(WETH_TOKEN).safeApprove(address(balVault), 0);
+        IERC20(WETH_TOKEN).safeApprove(address(balVault), type(uint256).max);
+        IERC20(BAL_ETH_POOL_TOKEN).safeApprove(address(balVault), 0);
+        IERC20(BAL_ETH_POOL_TOKEN).safeApprove(address(balVault), type(uint256).max);
+    }
+
+    /// @notice update the token to handler mapping
+    function _updateRewardToken(address _token, address _handler) internal {
+        rewardHandlers[_token] = _handler;
+    }
+
+    /// @notice Add a reward token and its handler
+    /// @dev For tokens that should not be swapped (i.e. BAL rewards)
+    ///      use address as zero handler
+    /// @param _token the reward token to add
+    /// @param _handler address of the contract that will sell for BAL or ETH
+    function addRewardToken(address _token, address _handler) external onlyOwner {
+        rewardTokens.push(_token);
+        _updateRewardToken(_token, _handler);
+    }
+
+    /// @notice Update the handler of a reward token
+    /// @dev Used to update a handler or retire a token (set handler to address 0)
+    /// @param _token the reward token to add
+    /// @param _handler address of the contract that will sell for BAL or ETH
+    function updateRewardToken(address _token, address _handler) external onlyOwner {
+        _updateRewardToken(_token, _handler);
```

Comment:
And do we want to sanity check a token is in rewardTokens before allowing this to be invoked. Possibly throw the check in the _updateRewardToken